module/executiondatasync/execution_data/execution_data_producer.go

```diff
@@ -0,0 +1,55 @@
+package execution_data
+
+import (
+	"errors"
+
+	"go.uber.org/atomic"
+
+	"github.com/onflow/flow-go/engine"
+)
+
+var ErrMultipleRegister = errors.New("producer may only be registered once")
+
+type ExecutionDataProducer interface {
+	Register(*engine.Notifier) error
+	SetLastProcessedHeight(uint64)
+	LastProcessedHeight() uint64
```

Comment:
How about this. the idea is that this is the highest height that this processor has completed processing. it's likely the last processed, but it's possible for blocks to be processed out of order (e.g. in the requester)
```suggestion
	HighestCompleteHeight() uint64
```

Reply:
[4b87f42](https://github.com/onflow/flow-go/pull/6217/commits/4b87f42a1f9693edeb7cfcc1c959532202400d71)